tools/patchwork_integration_test.sh

```diff
@@ -0,0 +1,102 @@
+#! /bin/bash
+set -e
+HOME="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+PUBIN_PATH=$1
+SINCE=$2
+MBOX=test-mbox.mbox
+PROJECT=test-patchwork
+INITIAL_IMPORT_SIZE=50 # Percentage of mbox to be imported as a raw mbox
+INITIAL_IMPORT_MBOX=initial.mbox
+
+clean_up() {
+	1>/dev/null 2>&1 docker stop pasta-patchwork
+}
+
+trap clean_up EXIT
+
+parse_pubin() {
+	git -C $PUBIN_PATH --no-pager log --pretty=format:%H --since=$SINCE --reverse | \
+		xargs -n 1 -I {} bash -c "
+				git -C $PUBIN_PATH --no-pager show --no-patch --pretty=format:'From %H %ad' {};
+				echo;
+				git -C $PUBIN_PATH --no-pager show {}:m;
+				echo -en '\n\n';
+			"
+}
+
+get_config() {
+	if [ $1 = "mbox" ]
+	then
+		initial_archive=$INITIAL_IMPORT_MBOX
+	else
+		initial_archive=""
+	fi
+	config=$(cat <<-END
+		[PaStA]
+		MODE = "mbox"
+		UPSTREAM = "HEAD^..HEAD"
+
+		[mbox]
+		[mbox.patchwork]
+		url='http://$container_addr:8000/api/'
+		projects = [{id=$project_id, initial_archive="$initial_archive", list_email="test@test.com"}]
+		END
+	)
+}
+
+cd "$HOME"
+
+echo "Parsing pubin..."
+parse_pubin > $MBOX
+if [ ! -d patchwork ]
+then
+	git clone https://github.com/getpatchwork/patchwork.git
+fi
+cp ./pasta-setup.sh patchwork/
+cp ./create_project.py patchwork/
+cp ./$MBOX patchwork/
+cd patchwork
+rm -f *.tar
+echo "UID=$UID" > .env
+echo "GID=`id -g`" >> .env
+docker-compose build
+docker-compose run --rm web ./pasta-setup.sh $MBOX $PROJECT
+
+n=$(( $(wc -l $PROJECT.mbox | cut -d" " -f 1) * $INITIAL_IMPORT_SIZE / 100 ))
+head -n $n $PROJECT.mbox > $INITIAL_IMPORT_MBOX
+mkdir -p ../../resources/linux/resources/mbox/patchwork/
+mv $INITIAL_IMPORT_MBOX $_
+
+echo Starting Patchwork web server...
+docker-compose run --rm --name pasta-patchwork web ./manage.py runserver 0.0.0.0:8000 &
+sleep 5 # Wait for container to start
```

Comment:
dangerous... What if the container isn't ready after 5s? Please wait for the availability of the container (e.g., dummy API call via cURL or sth like that).

Reply:
Yeah this is certainly hacky, okay I will try to poll the container using cURL